client_test.go

```diff
@@ -228,3 +229,12 @@ func TestClientZeroLengthPacket(t *testing.T) {
 		c.Close()
 	}
 }
+
+func TestClientShortPacket(t *testing.T) {
+	// init packet too short.
+	packet := []byte{0, 0, 0, 1, 2}
+
+	r := bytes.NewReader(packet)
+	_, err := NewClientPipe(r, &sink{})
+	require.Equal(t, errShortPacket, err)
```

Comment:
I don’t think we need to use `require.Equal` here, and in particular this usage should probably be “considered harmful”. Since this is essentially equivalent to `if err != errShortPacket { … }`.

While the `testify` packages do offer `require.EqualError` to make an additional assertion that it must be an error, it’s working through string compares, which is equally bad.

Instead, the simplest dumbest use is now overwhelmingly the best recommended practice:
```
if errors.is(err, errShortPacket) {
  …
}
```